### DISCUSSION TITLE
Disallow usage of `Sidekiq::Testing.inline!` without a block in specs

### DISCUSSION THREAD

**Discussion by Bennet-Sunder** [rubocop-hq/rubocop-rspec#1747] [Ideas]
From Sidekiq's testing guide, it is suggested to use `Sidekiq::Testing.inline!` with a block so that it doesn't share state. 

```
# or use block mode to avoid leaking the setting into other tests
Sidekiq::Testing.inline! do
  HardWorker.perform_async
  assert_worked_hard
end

```
https://github.com/sidekiq/sidekiq/wiki/Testing#testing-workers-inline

So this should throw a warning for the below spec

```
Sidekiq::Testing.inline!
HardWorker.perform_async
assert_worked_hard
```

**Comment by pirj** [comment]
We don’t keep cops that relate to testing third-party libraries in rubocop-rspec. See https://github.com/rubocop/rubocop-rspec/issues/1374
https://github.com/rubocop/rubocop-rspec/issues/1646
We’ve even extracted Capybara and FactoryBot, and plan on extracting rubocop-rspec_rails.
